File: datasets/IR-Plag/convertai-original-java-rest-kotlin/case-07/plagiarized/L4/04/Kasus7.kt
package IR

import java.util.Scanner

/**
 * @author C166AB8FBBBE55174E16A67530426A8A 19011F43894F06E2BA0CA74061427469
 */
fun inputMatrix(matrix: Array<DoubleArray>, n: Int) {
    val sc = Scanner(System.`in`)

    for (i in 0 until n) {
        for (j in 0 until n) {
            matrix[i][j] = sc.nextDouble()
        }
    }
}

fun sumMajorDiagonal(matrix: Array<DoubleArray>) {
    var sum = 0.0
    for (i in matrix.indices) {
        sum += matrix[i][i]
    }
    println("Sum of the elements in the major diagonal is $sum")
}

fun main(args: Array<String>) {
    val n = 4
    println("Enter a 4 by 4 matrix row by row: ")
    val matrix = Array(n) { DoubleArray(n) }

    inputMatrix(matrix, n)
    sumMajorDiagonal(matrix)
}

File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/plagiarized/L6/04/Kasus7.py
import numpy as np

def input_matrix(n):
    matrix = np.zeros((n, n))
    for i in range(n):
        for j in range(n):
            matrix[i, j] = float(input())
    return matrix

def sum_major_diagonal(matrix, n):
    if n >= 0:
        return matrix[n, n] + sum_major_diagonal(matrix, n - 1)
    else:
        return 0

if __name__ == "__main__":
    n = 4
    print("Enter a 4 by 4 matrix row by row:")
    matrix = input_matrix(n)
    print("Sum of the elements in the major diagonal is", sum_major_diagonal(matrix, n - 1))

